### PR TITLE
Badge condition rework to expand trigger and check options

### PR DESCRIPTION
**Pull request by yoshiweegee**
Uses the switch and variable caches to loop over all of the conditions for a badge without network overhead, only requesting syncs as needed.

### CODE REVIEW

File: server/handlers.go [hunk 1]
```diff
@@ -587,8 +587,31 @@ func (c *RoomClient) handleSs(msg []string) (err error) {
...
+						var condSwitchesPass bool = true;
+						for _, otherCondSwId := range condition.SwitchIds {
+							if otherCondSwId == switchId {
+								// this is the same switch just received, so it must be correct
+								continue
+							}
+							othCondSwVal, ok := c.switchCache[otherCondSwId]
+							if !ok {
+								// state of switch unknown; request from client and break
+								c.send <- buildMsg("ss", otherCondSwId, "0")
```
Comment (Flashfyre) [inline server/handlers.go:601]: This 'ss' is what guarantees that the cached value is up to date in all cases. This logic is hard to follow, but my understanding is that if a previously cached value of the switch is stored, it will not properly update if the trigger is anything other than blank (map trigger) which is the only case that calls ss and sv with type arg 1 or 2 in order to update the cached value every time the switch or variable updates. This would likely work for map trigger badges, but would often break others which are kicked off by a trigger and check switches and variables as a secondary condition.

**Review by Flashfyre — CHANGES_REQUESTED** [review]

**Comment by rumatsuki** [comment]
This coding style feels a lot more like C++ than it does Go. To fit in with the rest of the codebase please remove (most if not all) semicolons and shorten variable declarations like `var condSwitchesPass bool = true` to `condSwitchesPass := true`. Go infers type automatically, and if you have a value for something already then its best to do a short declaration.

**Comment by yoshiweegee** [comment]
I'll look into both of these. Thank you for the feedback!

File: server/handlers.go (hunk 1 above)
Comment (yoshiweegee) [reply server/handlers.go:601]: After learning the logic flow and considering possibilities, the current issue presented is that when a condition is "triggerless", it only instructs the client to update the first switch, or variable if `VarTrigger` is `true`, in the array. Changing the logic for all switches or variables to trigger a check possibly could break certain badge conditions that rely on this logic.
    If this isn't currently an issue, it seems that the most straightforward way to resolve this would be to push sync ("ss"/"sv" type >=1) all switches/variables, ensuring that the caches are always up to date. This would also deprecate `VarTrigger`. The downside is a lot of possible network overhead, since there's currently no way to instruct the client to stop sending updates about a particular switch or variable (other than map change).
    However, if we wanted to ensure that existing badge conditions don't break by supporting arbitrary combinations of triggers, I think changes to the `Condition` struct would be necessary.
    Ideally, supporting any combination of triggers--perhaps even between different types of triggers--would be the best solution, but it'd also be the most complex, likely requiring thorough review. What would be the best course of action for now?

Comment (yoshiweegee) [reply server/handlers.go:602]: (Resolving after decision to make changes to the Condition struct)
